lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1178,17 +1178,31 @@ def _process_logging_relation_changed(self, relation: Relation):
         Args:
             relation: the `Relation` instance to update.
         """
-        relation.data[self._charm.unit].update({"node_info": self._loki_push_api})
-        logger.debug("Saved Loki url in relation data %s", self._loki_push_api)
         if self._charm.unit.is_leader():
             relation.data[self._charm.app].update(self._promtail_binary_url)
             logger.debug("Saved promtail binary url: %s", self._promtail_binary_url)
+            relation.data[self._charm.app]["endpoints"] = json.dumps(self._endpoints())
+            logger.debug("Saved endpoints in relation data")
 
         if relation.data.get(relation.app).get("alert_rules"):
             logger.debug("Saved alerts rules to disk")
             self._remove_alert_rules_files(self.container)
             self._generate_alert_rules_files(self.container)
 
+    def _endpoints(self):
```

Comment:
Nice. I should probably do the same for alertmanager peer relation data.

```suggestion
    def _endpoints(self) -> List[dict]:
```